native/core/src/execution/planner.rs

```diff
@@ -878,6 +879,7 @@ impl PhysicalPlanner {
         return_type: Option<&spark_expression::DataType>,
         op: DataFusionOperator,
         input_schema: SchemaRef,
+        fail_on_overflow: bool,
```

Comment:
Maybe consider passing in the eval mode here instead of a boolean? We'll eventually need to support all three modes.